src/NCronJob/JobExecutionContext.cs

```diff
@@ -6,7 +6,20 @@ internal sealed record JobExecutionContext : IJobExecutionContext
 {
     internal bool ExecuteChildren = true;
 
-    public JobExecutionContext(JobRun jobRun) => JobRun = jobRun;
+    public JobExecutionContext(JobRun jobRun)
+    {
+        JobRun = jobRun;
+        if (jobRun.JobDefinition.IsStartupJob)
+        {
+            TriggerType = TriggerType.Startup;
+        }
+        else
+        {
+            TriggerType = jobRun.JobDefinition.CronExpression is not null
```

Comment:
@linkdotnet Nothing prevents a user from using the InstantJobRegistry to run a Job bearing a CronExpression so I believe checking this property may not be enough.

Reply:
Good catch

Reply:
Maybe should we work at the `JobRun.Create()` overloads?